azext_iot/common/credential_track_conversion.py

```diff
@@ -0,0 +1,40 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+from typing import Optional
+
+
+class Track1Credential:
+
+    def __init__(self, credential, resource):
+        """Track 1 credential that can be fed into Track 1 SDK clients. Exposes signed_session protocol.
+        Note: Cross-tenant authentication is not supported.
+
+        :param credential: Track 2 credential that exposes get_token protocol
+        :param resource: AAD resource
+        """
+        self._credential = credential
+        self._resource = resource
+
+    def signed_session(self, session=None):
+        import requests
+        from azure.cli.core.auth.util import resource_to_scopes
```

Comment:
`Profile.get_raw_token()` which the current IoT Hub/DPS oauth credential pattern is using already include this import and its usage. At the same time it includes more environments like MSI and cloud shell.

What is the benefit of this, versus using Profile.get_raw_token as we do in IoT Hub & DPS?

Reply:
jk - insert the friendship meme ended